src/memcache.c

```diff
@@ -733,9 +733,15 @@ PHP_MINIT_FUNCTION(memcache)
 
 	INIT_CLASS_ENTRY(ce, "MemcachePool", php_memcache_pool_class_functions);
 	memcache_pool_ce = zend_register_internal_class(&ce);
+#if PHP_VERSION_ID >= 80200
+	memcache_pool_ce->ce_flags |= ZEND_ACC_ALLOW_DYNAMIC_PROPERTIES;
```

Comment:
This looks like it'll work.

If the maintainers wanted to mark this as allowing dynamic properties, the attribute could optionally also be added to indicate that in reflection (include Zend/zend_attributes.h conditionally as well), though it isn't required

`zend_add_class_attribute(ast_node_ce, zend_ce_allow_dynamic_properties->name, 0);`

-----

If the maintainers went with declaring properties, then zend_declare_property_ex would be used
(e.g. if they wanted to warn users about typos, or didn't expect users to add any other dynamic properties in practice)

```c
	zend_declare_property_ex(memcache_pool_ce, name, value, ZEND_ACC_PUBLIC, NULL)
```